aten/src/ATen/native/mps/OperationUtils.h

```diff
@@ -0,0 +1,196 @@
+//  Copyright © 2022 Apple Inc.
+
+#include <ATen/ATen.h>
+#include <ATen/Tensor.h>
+#include <ATen/Utils.h>
+#include <ATen/mps/MPSStream.h>
+#include <ATen/native/mps/TensorFactory.h>
+#include <c10/core/ScalarType.h>
+#include <torch/library.h>
+#include <unordered_map>
+
+#ifdef __OBJC__
+#include <MetalPerformanceShaders/MetalPerformanceShaders.h>
+#endif
+
+using namespace at::mps;
+
+namespace at {
+namespace native {
+namespace mps {
+
+struct TORCH_CUDA_CPP_API MPSGeneratorImpl : public c10::GeneratorImpl {
+  MPSGeneratorImpl(DeviceIndex device_index = -1);
+  ~MPSGeneratorImpl() = default;
+
+  void set_current_seed(uint64_t seed) override;
+  uint64_t current_seed() const override;
+  uint64_t seed() override;
+  void set_state(const c10::TensorImpl& new_state) override;
+  c10::intrusive_ptr<c10::TensorImpl> get_state() const override;
+  static DeviceType device_type();
+
+private:
+  MPSGeneratorImpl* clone_impl() const override;
+  uint64_t seed_ = default_rng_seed_val;
```

Comment:
you're probably going to need to pick an RNG algorithm for MPS. Does Philox work?

Reply:
Yes. We use Philox.